components/ActivityCardPreview/index.js

```diff
@@ -0,0 +1,58 @@
+import Image from "next/image";
+import styled from "styled-components";
+import getFilteredCategories from "@/utils/filterCategories";
+import Link from "next/link";
+import BookmarkButton from "../BookmarkButton";
+
+export const StyledCateogriesUl = styled.ul`
+  list-style: none;
+  padding: 5px;
+  display: flex;
+  justify-content: center;
+  gap: 10px;
+`;
+
+export const StyledCategory = styled.li`
+  border: 1px solid grey;
+  padding: 5px;
+  border-radius: 5px;
+  margin: 10px 0px;
+`;
+
+const StyledImage = styled(Image)`
+  width: auto;
+  height: auto;
+`;
+
+export default function ActivityCardPreview({
+  activity,
+  isBookmarked,
+  onToggleBookmark,
+}) {
+  const { title, categoryIds, imageUrl } = activity;
+  const filterCategories = getFilteredCategories(categoryIds);
```

Comment:
I can't see any benefit for changing the category strings into objects. Since each category is unique you can use it as a key in the map function. Instead now you do a lot of extra work. But this is up to you... 😊

Reply:
We did this with Merle, because we had an error with Vercel and thought this might be a solution and less confusing. So until it causes further problems we will leave it like that. :)